scripts/plotting/makeDataMCStackPlot.py

```diff
@@ -0,0 +1,67 @@
+from wremnants.datasets.datagroups import datagroups2016
+from wremnants import histselections as sel
+from wremnants import plot_tools
+import matplotlib.pyplot as plt
+import argparse
+import os
+import shutil
+import logging
+import pathlib
+
+template_dir = f"{pathlib.Path(__file__).parent}/Template"
+
+parser = argparse.ArgumentParser()
+parser.add_argument("infile", help="Output file of the analysis stage, containing ND boost histograms")
+parser.add_argument("--wlike", action='store_true', help="Make W like plots")
+parser.add_argument("-n", "--baseName", type=str, help="Histogram name in the file (e.g., 'nominal')", default="nominal")
+parser.add_argument("--addUncorrected", action='store_true', help="Add uncorrected (no N3LL) distribution to plots")
+parser.add_argument("--hists", type=str, nargs='+', help="List of histograms to plot")
+parser.add_argument("-c", "--channel", type=str, choices=["plus", "minus", "all"], default="all", help="Select channel to plot")
+parser.add_argument("-p", "--outpath", type=str, default=os.path.expanduser("~/www/WMassAnalysis"), help="Base path for output")
+parser.add_argument("-f", "--outfolder", type=str, default="test", help="Subfolder for output")
+parser.add_argument("-r", "--rrange", type=float, nargs=2, default=[0.9, 1.1], help="y range for ratio plot")
+args = parser.parse_args()
+
+groups = datagroups2016(args.infile, wlike=args.wlike)
+histInfo = groups.datagroupsForHist(args.baseName, syst="")
+if args.addUncorrected:
+	histInfo = groups.datagroupsForHist(args.baseName, syst="uncorr")
+	histInfo = groups.addUncorrectedProc(args.baseName, "uncorr")
+
+prednames = [x for x in histInfo.keys() if x not in ["Data", "uncorr"]]
+select = {} if args.channel == "all" else {"select" : -1.j if args.channel == "minus" else 1.j}
+
+outpath = "/".join([args.outpath, args.outfolder])
+if not os.path.isdir(args.outpath):
+	raise IOError(f"The path {args.outpath} doesn't not exist. You should create it (and possibly link it to your web area)")
+	
+if not os.path.isdir(outpath):
+	logging.info(f"Creating folder outpath")
+	os.makedirs(outpath)
+
+xlabels = {
+	"pt" : r"p$_{T}^{\ell}$ (GeV)",
+	"eta" : r"\eta^{\ell}$",
+	"unrolled" : r"(p$_{T}^{\ell}$, $\eta^{\ell}$) bin",
+}
+
+scales = {
+	"pt" : 9e6 if not args.wlike else 2e6,
+	"eta" : 5e6 if not args.wlike else 2e5,
+	"unrolled" : 1.8e5 if not args.wlike else 1.5e4,
+}
```

Comment:
Do these really need to be hardcoded?

Reply:
Ok changed this to use the autoscaling by default, with optional arg for scale